commands/http_client.go

```diff
@@ -59,3 +66,40 @@ func (c *HTTPClient) GetSupportBundleStatus(bundleID bundleID) (status int, resp
 	}
 	return resp.StatusCode, responseBytes, nil
 }
+
+type SupportBundleCreationOptions struct {
+	Name        string                   `json:"name"`
+	Description string                   `json:"description"`
+	Parameters  *SupportBundleParameters `json:"parameters"`
+}
+
+type SupportBundleParameters struct {
+	Configuration bool                               `json:"configuration"`
+	Logs          *SupportBundleParametersLogs       `json:"logs,omitempty"`
+	System        bool                               `json:"system"`
+	ThreadDump    *SupportBundleParametersThreadDump `json:"thread_dump"`
+}
+
+type SupportBundleParametersLogs struct {
+	Include   bool   `json:"include"`
+	StartDate string `json:"start_date"`
+	EndDate   string `json:"end_date"`
+}
+
+type SupportBundleParametersThreadDump struct {
+	Count    uint `json:"count"`
+	Interval uint `json:"interval"`
+}
+
+func (p SupportBundleCreationOptions) MarshalJSON() ([]byte, error) {
```

Comment:
All this because if you don't send an empty json object the sb creation fails?

Reply:
Exactly